### PR TITLE
Feature/docker integration

### PR DESCRIPTION
**Pull request by fadyawan**

### CODE REVIEW

File: express-app/index.js [hunk 1]
```diff
@@ -1,10 +1,66 @@
...
+        }
+    }
+    if (userExist) {
+        console.log('error, user already exists')
+        res.status(400).send(`${username} has already been registered`)
+        console.log(`${username} has already been registered`)
+    }
+    else {
+        const createUser = {};
+        createUser.username = username
```
Comment (Ripraskel) [inline express-app/index.js]: You could write something like this: 
const new User = {
  username, // same as username: username
  password
}

File: express-app/index.js [hunk 2]
```diff
@@ -1,10 +1,66 @@
...
+        createUser.password = password
+        users.push(createUser)
+        res.send(`${username} has registered succesfully`)
+        console.log(`${username} has been registered`)
+    }
+})
+
+app.post('/auth/login', (req, res) => {
+    const username = req.body.username;
```
Comment (Ripraskel) [inline express-app/index.js]: A standard practice for this is to do something called destructuring:
const { username, password } = req.body;

File: express-app/package.json [hunk 3]
```diff
@@ -10,6 +10,9 @@
   "author": "",
   "license": "ISC",
   "dependencies": {
+    "body-parser": "^1.20.2",
+    "cors": "^2.8.5",
+    "express": "^4.18.2",
     "nodemon": "^2.0.22"
```
Comment (Ripraskel) [inline express-app/package.json:16]: Are you using nodemon?

File: react-app/package.json [hunk 4]
```diff
@@ -10,6 +10,7 @@
     "@testing-library/jest-dom": "^5.16.5",
```
Comment (Ripraskel) [inline react-app/package.json]: These @testing-library should be dev dependencies

File: react-app/src/Pages/Register Page/RegisterPage.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
+import './RegisterPage.css';
+import NavBar from '../../Components/navBar/navBar';
+import { TextField, Typography, Button } from '@mui/material';
+import { useState } from 'react';
+import axios from 'axios';
+
+function LoginPage() {
```
Comment (Ripraskel) [inline react-app/src/Pages/Register Page/RegisterPage.js]: RegisterPage?

File: react-app/src/Pages/Register Page/RegisterPage.js [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+                        variant="standard" 
+                        required
+                    />
+                    <br />
+                    <br />
+                    <Typography variant="body1">Password</Typography>
+                    <TextField 
+                        id="standard-basic"
+                        onChange={(e) => setPassword(e.target.value)}
+                        variant="standard"
```
Comment (Ripraskel) [inline react-app/src/Pages/Register Page/RegisterPage.js]: Is there a password variant to hide characters?

File: react-app/src/Pages/Register Page/RegisterPage.js [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+                    <br />
+                    <br />
+                    <Typography variant="body1">Password</Typography>
+                    <TextField 
+                        id="standard-basic"
+                        onChange={(e) => setPassword(e.target.value)}
+                        variant="standard" 
+                        required
+                    />
+                    <Button variant="contained" onClick={login}>Log In</Button>
```
Comment (Ripraskel) [inline react-app/src/Pages/Register Page/RegisterPage.js]: Does this on click actually do anything? Login function not defined above that I can see.

File: react-app/src/Pages/Register Page/RegisterPage.js [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+import NavBar from '../../Components/navBar/navBar';
+import { TextField, Typography, Button } from '@mui/material';
+import { useState } from 'react';
+import axios from 'axios';
+
+function LoginPage() {
+    const [username, setUsername] = useState('')
+    const [password, setPassword] = useState('')
+
+    const submitDetails = (e) => {
```
Comment (Ripraskel) [inline react-app/src/Pages/Register Page/RegisterPage.js]: Is this function obsolete?

File: react-app/src/Pages/Register Page/RegisterPage.js [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+
+        if (username && password) {
+            console.log(username, password)
+        }
+        else {
+            console.log('error')
+        }
+    }
+
+    const registerDetails = async(e) => {
```
Comment (Ripraskel) [inline react-app/src/Pages/Register Page/RegisterPage.js]: Try catch on different branch?

File: express-app/index.js [hunk 10]
```diff
@@ -1,10 +1,66 @@
...
+const cors = require('cors');
+
+app.use(cors())
+app.use(bodyParser.json());
+
+const users = [];
+
+app.post('/auth/register', (req, res) => {
+    console.log(users)
+    const username = req.body.username;
```
Comment (Ripraskel) [inline express-app/index.js]: Same as comment below about destructruring

File: express-app/Dockerfile [hunk 11]
```diff
@@ -0,0 +1,12 @@
+FROM node:18-alpine
+
+WORKDIR /app
+
+COPY . .
```
Comment (Ripraskel) [inline express-app/Dockerfile:7]: As you did for react best to copy package.json then run the install then copy the rest of the files before running.

**Review by Ripraskel — CHANGES_REQUESTED** [review]
Some things to look at but good work.

File: react-app/src/Pages/Register Page/RegisterPage.js (hunk 7 above)
Comment (fadyawan) [reply react-app/src/Pages/Register Page/RegisterPage.js]: I was going to separate the login and register pages.

File: react-app/src/Pages/Register Page/RegisterPage.js (hunk 8 above)
Comment (fadyawan) [reply react-app/src/Pages/Register Page/RegisterPage.js]: Yes

File: react-app/src/Pages/Register Page/RegisterPage.js (hunk 9 above)
Comment (fadyawan) [reply react-app/src/Pages/Register Page/RegisterPage.js]: This is an older version of the login page that I duplicated for the register page.

**Pull request merged**